components/layout/Auth/TogglePassword.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { useState } from "react";
+import visibilityOn from "@/public/images/icons/btn_visibility_on.svg";
+import visibilityOff from "@/public/images/icons/btn_visibility_off.svg";
+import Image from "next/image";
+import styles from "./Auth.module.scss";
+
+interface Props {
+  onPasswordVisible: (isVisible: boolean) => void;
+}
+
+const TogglePassword: React.FC<Props> = ({ onPasswordVisible }) => {
+  const [isVisible, setVisible] = useState(false);
+  const imgToggle = isVisible ? visibilityOn : visibilityOff;
+
+  const handleClick = () => {
+    const updatedVisible = !isVisible;
+    setVisible(updatedVisible);
+    onPasswordVisible(updatedVisible);
+  };
+
```

Comment:
주소의 변경에 따라 isVisible 기본 값을 false로 설정해주는 로직이 필요합니다.
signup과 login 주소를 오가는 과정에서 password 에 해당하는 TogglePassword 컴포넌트는 최초랜더링 이후 리랜더링만 진행되고 있습니다. 이 과정에서 이전에 선택한 상태값이 보존되고 있으므로 페이지가 변화하였을 때는 기본 상태를 변경하는 로직이 필요합니다. 😁

```suggestion
  const router = useRouter();
  useEffect(() => {
    setVisible(false);
  }, [router.asPath]);
```

Reply:
@myong39